tests/sentry/tasks/test_backfill_seer_grouping_records.py

```diff
@@ -1754,53 +1764,73 @@ def test_backfill_seer_grouping_records_skip_project_already_processed(self, moc
         assert mock_logger.info.call_args_list == expected_call_args_list
 
     @with_feature("projects:similarity-embeddings-backfill")
+    @override_options({"similarity.backfill_total_worker_count": 1})
+    @override_options({"similarity.new_project_seer_grouping.enabled": True})
     @patch("sentry.tasks.embeddings_grouping.backfill_seer_grouping_records_for_project.logger")
     @patch("sentry.tasks.embeddings_grouping.utils.post_bulk_grouping_records")
-    def test_backfill_seer_grouping_records_reprocess_project_already_processed(
+    def test_backfill_seer_grouping_records_typical_backfill_request(
         self, mock_post_bulk_grouping_records, mock_logger
     ):
         """
-        Test that projects that have a backfill completed project option are not skipped when not
-        passed the skip_processed_projects flag.
+        Test that projects that have the backfill completed option set are skipped when we backfill
+        all projects.
         """
+        # Create two more projects and one of them is already backfilled
+        project2 = self.create_project(organization=self.organization)
+        project2.update_option(PROJECT_BACKFILL_COMPLETED, int(time.time()))
+        project3 = self.create_project(organization=self.organization)
+
         mock_post_bulk_grouping_records.return_value = {"success": True, "groups_with_neighbor": {}}
-        self.project.update_option("sentry:similarity_backfill_completed", int(time.time()))
         with TaskRunner():
-            backfill_seer_grouping_records_for_project(self.project.id, None)
+            # This is the typical way of backfilling all projects
+            backfill_seer_grouping_records_for_project(
+                current_project_id=None,
+                last_processed_group_id_input=None,
+                skip_processed_projects=True,
+                enable_ingestion=True,
+                worker_number=0,
+            )
 
+        key = "backfill_seer_grouping_records"
+        # Since we set the total worker count to 1, the project cohort will have all projects
+        # except the one that has already been backfilled
+        cohort = [self.project.id, project3.id]
```

Comment:
if we want to test that `skip_processed_projects=True` works as intended, shouldn't we include `project2`, the project that is already processed, in the cohort and have the test check that it's not being re-processed?

Reply:
thanks for following up on slack!